tests/static_analyzer/test_lsp_client.py

```
@@ -1068,3 +1070,56 @@ def test_shutdown_clears_document_tracking(self):
+        with patch("static_analyzer.engine.lsp_client.process_tree_rss", return_value=12_479 * 1024 * 1024):
+            with pytest.raises(TimeoutError) as excinfo:
+                client._send_request("textDocument/documentSymbol", {}, timeout=1)
+
+        message = str(excinfo.value)
+        assert "textDocument/documentSymbol" in message
+        assert "12.2 GB" in message or "GB" in message
+
+    def test_timeout_includes_server_stderr(self):
+        client, _ = self._client_with_server(["Roslyn.Solution: loading", "Out of memory."])
+        with patch("static_analyzer.engine.lsp_client.process_tree_rss", return_value=0):
+            with pytest.raises(TimeoutError, match="Out of memory."):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)